common/time-service/src/lib.rs

```diff
@@ -170,6 +200,17 @@ pub trait TimeServiceTrait: Send + Sync + Clone + Debug {
         Interval::new(delay, period)
     }
 
+    /// Creates a new [`Interval`] that yields with interval of `period`. The
+    /// first tick completes after the `start` deadline. An interval will tick
+    /// indefinitely.
+    ///
+    /// See [`interval`](#method.interval) for more details.
+    #[cfg(any(test, feature = "async"))]
+    fn interval_at(&self, start: Instant, period: Duration) -> Interval {
+        let delay = self.sleep_until(start);
+        Interval::new(delay, period)
+    }
+
```

Comment:
I'm assuming `timeout_at` and `interval_at` are for future changes?  Otherwise, it's not really directly `Instant` support, except I guess it uses it.

Reply:
`interval_at` is sort of orthogonal, I agree. I figured I would just 100% replicate the full tokio while I'm here. `timeout_at` is related though, since it takes an `Instant` deadline instead of a relative duration.